lightningd/peer_control.c

```diff
@@ -692,33 +709,60 @@ static void json_add_channel(struct lightningd *ld,
 				   channel->channel_info.their_config.channel_reserve,
 				   "our_channel_reserve_satoshis",
 				   "our_reserve_msat");
-	/* Compute how much we can send via this channel. */
+
+
+	/* Compute how much we can send via this channel in one payment. */
 	if (!amount_msat_sub_sat(&spendable,
 				 channel->our_msat,
 				 channel->channel_info.their_config.channel_reserve))
 		spendable = AMOUNT_MSAT(0);
-
 	/* Take away any currently-offered HTLCs. */
 	subtract_offered_htlcs(channel, &spendable);
-
 	/* If we're funder, subtract txfees we'll need to spend this */
 	if (channel->funder == LOCAL) {
 		if (!amount_msat_sub_sat(&spendable, spendable,
 					 commit_txfee(channel, spendable)))
 			spendable = AMOUNT_MSAT(0);
 	}
-
 	/* We can't offer an HTLC less than the other side will accept. */
 	if (amount_msat_less(spendable,
 			     channel->channel_info.their_config.htlc_minimum))
 		spendable = AMOUNT_MSAT(0);
-
 	/* We can't offer an HTLC over the max payment threshold either. */
 	if (amount_msat_greater(spendable, chainparams->max_payment))
 		spendable = chainparams->max_payment;
-
+	/* append spendable to JSON output */
 	json_add_amount_msat_compat(response, spendable,
 				    "spendable_msatoshi", "spendable_msat");
+
+
+	/* Compute how much we can receive via this channel in one payment */
+	if (!amount_sat_sub_msat(&their_msat, channel->funding, channel->our_msat))
+		their_msat = AMOUNT_MSAT(0);
+	if (!amount_msat_sub_sat(&receivable,
+				 their_msat,
+				 channel->our_config.channel_reserve))
+		receivable = AMOUNT_MSAT(0);
+	/* Take away any currently-offered HTLCs. */
+	subtract_received_htlcs(channel, &receivable);
+	/* If they're funder, subtract txfees they'll need to spend this */
+	if (channel->funder == REMOTE) {
+		if (!amount_msat_sub_sat(&receivable, receivable,
+					 commit_txfee(channel, receivable)))
```

Comment:
Hm, I'm sure we need to modify `commit_txfee(channel, spendable)` or add an analogue function that calculates this for the remote side...Will work something out...  The unit tests didn't find this, since fee configs are the same for every testtnode...

Reply:
Im reviewing myself here, please wait until I addressed this one :D

Reply:
addressed by 99948c9cf